frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditAttachment/EditAttachment.tsx

```diff
@@ -0,0 +1,197 @@
+
+import { AttachmentStackedBar } from './AttachmentStackedBar'
+
+type EditAttachmentProps = EditFieldProps<AttachmentFieldBase>
+
+const EDIT_ATTACHMENT_FIELD_KEYS = [
+  'title',
+  'description',
+  'required',
+  'attachmentSize',
+] as const
+
+type EditAttachmentKeys = typeof EDIT_ATTACHMENT_FIELD_KEYS[number]
+
+type EditAttachmentInputs = Pick<AttachmentFieldBase, EditAttachmentKeys>
+
+const transformAttachmentFieldToEditForm = (
+  field: AttachmentFieldBase,
+): EditAttachmentInputs => {
+  return pick(field, EDIT_ATTACHMENT_FIELD_KEYS)
+}
+
+const transformAttachmentEditFormToField = (
+  inputs: EditAttachmentInputs,
+  originalField: AttachmentFieldBase,
+): AttachmentFieldBase => {
+  return extend({}, originalField, inputs)
+}
+
+export const EditAttachment = ({ field }: EditAttachmentProps): JSX.Element => {
+  const { data: form } = useCreateTabForm()
+  const {
+    register,
+    formState: { errors },
+    control,
+    getValues,
+    isSaveEnabled,
+    buttonText,
+    handleUpdateField,
+    isLoading,
+    handleCancel,
+  } = useEditFieldForm<EditAttachmentInputs, AttachmentFieldBase>({
+    field,
+    transform: {
+      input: transformAttachmentFieldToEditForm,
+      output: transformAttachmentEditFormToField,
+    },
+  })
+
+  const requiredValidationRule = useMemo(
+    () => createBaseValidationRules({ required: true }),
+    [],
+  )
+
+  const otherAttachmentsSize = useMemo(() => {
+    if (!form?.form_fields) return 0
+    return form.form_fields
+      .filter(
+        (ff): ff is FormFieldDto<AttachmentFieldBase> =>
+          ff.fieldType === BasicField.Attachment && ff._id !== field._id,
+      )
+      .reduce((sum, ff) => sum + Number(ff.attachmentSize), 0)
+  }, [field._id, form?.form_fields])
+
+  const maxTotalSizeMb: number = useMemo(() => {
+    if (!form?.responseMode) return 0
+    switch (form.responseMode) {
+      case FormResponseMode.Email:
+        return 7
+      case FormResponseMode.Encrypt:
+        return 20
+    }
+  }, [form?.responseMode])
+
+  const attachmentSizeOptions: ComboboxItem[] = useMemo(() => {
+    if (!form) return []
+    switch (form.responseMode) {
+      case FormResponseMode.Email:
+        return [
+          { value: AttachmentSize.OneMb, label: `${AttachmentSize.OneMb} MB` },
+          { value: AttachmentSize.TwoMb, label: `${AttachmentSize.TwoMb} MB` },
+          {
+            value: AttachmentSize.ThreeMb,
+            label: `${AttachmentSize.ThreeMb} MB`,
+          },
+          {
+            value: AttachmentSize.FourMb,
+            label: `${AttachmentSize.FourMb} MB`,
+          },
+          {
+            value: AttachmentSize.SevenMb,
+            label: `${AttachmentSize.SevenMb} MB`,
+          },
+        ]
+      case FormResponseMode.Encrypt:
+        return Object.values(AttachmentSize).map((size) => ({
+          value: size,
+          label: `${size} MB`,
+        }))
+    }
```

Comment:
Not sure if Typescript will like it, but I'd share some of the formatting code here like this:

```suggestion
    let sizes;
    
    if (form.responseMode === FormResponseMode.Email) {
        sizes = [
          AttachmentSize.OneMb,
          AttachmentSize.TwoMb,
          AttachmentSize.ThreeMb,
          AttachmentSize.FourMb,
          AttachmentSize.SevenMb
        ]
    } else if (form.responseMode === FormResponseMode.Encrypt) {
      sizes = Object.values(AttachmentSize);
    }
    
    return sizes.map((size) => ({
        value: size,
        label: `${size} MB`,
      }))
```

Reply:
d869055